### PR TITLE
Get Tests Passing

### PR DESCRIPTION
**Pull request by AbuMareBear**
This PR gets our tests green. It removes many auto-generated tests that were not useful. It removes fixtures that were not being used. A passing test suite is critical for feature development.

### CODE REVIEW

File: Gemfile.lock [hunk 1]
```diff
@@ -347,6 +352,7 @@ GEM
     zeitwerk (2.6.12)
 
 PLATFORMS
+  arm64-darwin-23
```
Comment (AbuMareBear) [inline Gemfile.lock:355]: This diff is because I'm on a Mac with an M1 chip.

File: test/controllers/posts_controller_test.rb [hunk 2]
```diff
@@ -6,16 +6,24 @@ class PostsControllerTest < ActionDispatch::IntegrationTest
   end
 
   test "should get index" do
+    sign_in users(:abdullah)
+
     get posts_url
+
     assert_response :success
```
Comment (AbuMareBear) [inline test/controllers/posts_controller_test.rb:13]: To make this test pass it needed a signed in user. I also put spaces between the three phases of testing

1. setup
2. execution
3. verify

File: test/fixtures/profiles.yml [hunk 3]
```diff
@@ -1,9 +1,2 @@
...
-
-one:
-  nickname: MyString
-  profile_picture: MyString
-
-two:
-  nickname: MyString
-  profile_picture: MyString
+abdullah:
+  user: abdullah
```
Comment (AbuMareBear) [inline test/fixtures/profiles.yml:2]: I love fixtures. In my experience, they work better when they are an approximation of real world data.

File: test/test_helper.rb [hunk 4]
```diff
@@ -4,6 +4,8 @@
 
 module ActiveSupport
   class TestCase
+    include Devise::Test::IntegrationHelpers
```
Comment (AbuMareBear) [inline test/test_helper.rb:7]: This imports the devise `sign_in` method that is used in the tests.

File: Gemfile.lock (hunk 1 above)
Comment (odilson-dev) [reply Gemfile.lock:355]: Okay

File: test/controllers/posts_controller_test.rb (hunk 2 above)
Comment (odilson-dev) [reply test/controllers/posts_controller_test.rb:13]: Great!

File: test/fixtures/profiles.yml (hunk 3 above)
Comment (odilson-dev) [reply test/fixtures/profiles.yml:2]: Make sense

File: test/test_helper.rb (hunk 4 above)
Comment (odilson-dev) [reply test/test_helper.rb:7]: Okay

**Review by odilson-dev — APPROVED** [review]

**Pull request merged**